src/app/service/event-template.service.ts

```diff
@@ -0,0 +1,116 @@
+import { Injectable } from '@angular/core';
+import { environment } from '../../environments/environment';
+import { BehaviorSubject, catchError, map, Observable, of } from 'rxjs';
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { EventTemplate } from '../entity/EventTemplate';
+import EntityState from '../entity/EntityState';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class EventTemplateService {
+  private data: EntityState<EventTemplate[]> = {
+    entity: [],
+    loading: false,
+    error: null,
+  };
+  private apiUrl: string = environment.baseUrl + '/v1/event-templates';
+  eventTemplateSubject = new BehaviorSubject<EntityState<EventTemplate[]>>(
+    this.data,
+  );
+
+  constructor(private _http: HttpClient) {
+    this.fetchAll();
+  }
+
+  getData(): Observable<EntityState<EventTemplate[]>> {
+    return this.eventTemplateSubject.asObservable();
+  }
+
+  fetchAll() {
+    this.data.loading = true;
```

Comment:
Why does `data` need to be a class member variable?

The issue with it being a class variable is the case when you ave multiple components using this service and making requests at the same time (since the service is a singleton)